planner/core/integration_test.go

```diff
@@ -1564,3 +1565,25 @@ func (s *testIntegrationSuite) TestDeleteUsingJoin(c *C) {
 	tk.MustQuery("select * from t1").Check(testkit.Rows("1 1"))
 	tk.MustQuery("select * from t2").Check(testkit.Rows("2 2"))
 }
+
+func (s *testIntegrationSerialSuite) Test19942(c *C) {
+	collate.SetNewCollationEnabledForTest(true)
+	defer collate.SetNewCollationEnabledForTest(false)
+
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("SET @@tidb_enable_clustered_index=1;")
+	tk.MustExec("CREATE TABLE test.`t` (" +
```

Comment:
I suggest changing the test to:

```
DROP TABLE IF EXISTS `test`.`t`;
CREATE TABLE test.`t` (
    `a` int(11) NOT NULL,
    `b` varchar(10) COLLATE utf8_general_ci NOT NULL,
    `c` varchar(50) COLLATE utf8_general_ci NOT NULL,
    `d` char(10) NOT NULL,
    PRIMARY KEY (`c`),
    UNIQUE KEY `a_uniq` (`a`),
    UNIQUE KEY `b_uniq` (`b`),
    UNIQUE KEY `d_uniq` (`d`),
    KEY `a_idx` (`a`),
    KEY `b_idx` (`b`),
    KEY `d_idx` (`d`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci;

INSERT INTO test.t (a, b, c, d) VALUES (1, '1', '0', '1');
INSERT INTO test.t (a, b, c, d) VALUES (2, ' 2', ' 0', ' 2');
INSERT INTO test.t (a, b, c, d) VALUES (3, '  3 ', '  3 ', '  3 ');
INSERT INTO test.t (a, b, c, d) VALUES (4, 'a', 'a   ', 'a');
INSERT INTO test.t (a, b, c, d) VALUES (5, ' A  ', ' A   ', ' A  ');
INSERT INTO test.t (a, b, c, d) VALUES (6, ' E', 'é        ', ' E');

SELECT * FROM `test`.`t` FORCE INDEX(`a_uniq`);
SELECT * FROM `test`.`t` FORCE INDEX(`b_uniq`);
SELECT * FROM `test`.`t` FORCE INDEX(`d_uniq`);
SELECT * FROM `test`.`t` FORCE INDEX(`a_idx`);
SELECT * FROM `test`.`t` FORCE INDEX(`b_idx`);
SELECT * FROM `test`.`t` FORCE INDEX(`d_idx`);
```

So that the "Padding" and unique key are both covered.